### ISSUE TITLE
Avoiding having to run terraform post boot.

### ISSUE THREAD

**Issue by Sadi-a** [poseidon/typhoon#1316]
### Description 

Currently, Typhoon uses SSH to give the workers and controllers some secrets and other files like certificates for example. 
Thus, I would like to make it so SSH is not needed to give secrets and other required files with flatcar-linux on bare metal. One way to do this would be to add an option which would allow one to include these files in the ignition.json file's definition. That way, SSH wouldn't be needed anymore and only the matchbox server would work on it.

### Motivation

When a node reboots, a human intervention is required to run `terraform apply`. With the addition of this feature, it wouldn't be necessary to run this command and the node would be ready immediately without waiting for a `terraform apply`.

**Comment by dghubble** [comment]
Duplicate https://github.com/poseidon/typhoon/issues/1026#issuecomment-893663674

tldr: SSH is used to securely distribute cluster credentials. PXE booting machines are by definition without identity, so you won't be able to devise an authentication mechanism for it without hardware attestation (a big topic). Exposing credentials to anyone on the network is not a viable approach.

> When a node reboots, a human intervention is required to run terraform apply

There is no need to re-run terraform apply whenever a node reboots.

**Comment by Snaipe** [comment]
I don't think what you said in your linked comment is quite true; you're omitting the existence of client certificates for authentication, which we've used with no issues to secure our ipxe chainloading.

The main use-case we have here is bringing up and down ephemeral workers. We have dozens of servers being used as workers in our build farm, and the way it works is that rebooting reprovisions (and wipes) the host. We're currently trying to consolidate our setup to avoid rolling everything that makes a functioning k8s distribution from scratch, and Typhoon seems to do a good job at that, except for the part that we can't really replace these live-booting nodes.

Note that this isn't necessarily purely bare-metal either; we do run Flatcar directly on bare-metal, but we're also exploring using the same setup on scalable worker pools using virtual machines configured to HTTP-boot. In all use-cases, ipxe is provisioned out of band with an embedded client certificate, which can be used to secure communication with the matchbox server. Having secrets on there isn't a problem.

But to your point, maybe what we're trying to do here is fundamentally orthogonal to what Typhoon is trying to accomplish. We will continue development of that direction in our own fork.

**Comment by dghubble** [comment]
So you've devised a machine authentication scheme where PXE booting machines use client certificates embedded in iPXE to authenticate with other services. If TFTP (or similar) servers serve iPXE to any device on the network (as is generally the case), this client certificate is available to anyone on the network. Its not really for suitable for authentication. Alternately, maybe your TFTP server requires clients to authentcate _somehow_ before serving the iPXE firmware. In that case, the authentication challenge is just moved lower in the stack and the question is how that pre-iPXE machine authenticates (simple MAC matching won't cut it, you'd need TPM attestation with registered PCRS)

Finally, Matchbox is [not](https://matchbox.psdn.io/deployment/#operational-notes) a suitable store for secret materials.

By constrast, SSH is a praactical and secure  approach to distribute credentials to cluster nodes when the cluster is created. I understand folks at Arista want ephemeral workers, but I can't just ignore the security problems with your proposal. If you must hand out credentials insecurely, output the assets or kubeconfig to some remote store and load them via a systemd unit or put them directly in your Ignition, disable SSH in your fork, and hope nobody finds your Ignition. But this can't be part of Typhoon.

**Comment by dghubble** [comment]
Also, I'd recommend Arista discuss needs via support@psdn.io, rather than opening a bunch of tickets in my issue tracker.
